sdk/identity/src/token_credentials/specific_azure_credential.rs

```diff
@@ -0,0 +1,340 @@
+use super::options;
+#[cfg(not(target_arch = "wasm32"))]
+use crate::AzureCliCredential;
+#[cfg(feature = "client_certificate")]
+use crate::ClientCertificateCredential;
+use crate::{
+    AppServiceManagedIdentityCredential, ClientSecretCredential, EnvironmentCredential,
+    TokenCredentialOptions, VirtualMachineManagedIdentityCredential, WorkloadIdentityCredential,
+};
+use azure_core::{
+    auth::{AccessToken, TokenCredential},
+    error::{ErrorKind, ResultExt},
+    Error,
+};
+use std::sync::Arc;
+
+pub const AZURE_CREDENTIAL_KIND: &str = "AZURE_CREDENTIAL_KIND";
+
+pub mod azure_credential_kinds {
+    pub const ENVIRONMENT: &str = "environment";
+    #[cfg(not(target_arch = "wasm32"))]
+    pub const AZURE_CLI: &str = "azurecli";
```

Comment:
I've not read this whole PR yet, but wanted to call out that the azureauth_cli_credential has been left out here, and since it's behind a feature flag and isn't building was broken in version 0.18, I'm in the process of opening an issue + PR, but want o make sure I take in all the context from this PR first :).

Reply:
It can be added, but I'm leaving that as a follow-up by us.